cpp/src/io/parquet/experimental/page_index_filter.cu

```diff
@@ -782,60 +782,49 @@ std::vector<std::vector<bool>> aggregate_reader_metadata::compute_data_page_mask
     thrust::counting_iterator<size_t>(0),
     thrust::counting_iterator(num_columns),
     [&](auto const col_idx) {
-      data_page_mask_tasks.emplace_back(
-        cudf::detail::host_worker_pool().submit_task([&, col_idx = col_idx] {
-          // Construct a row indices mapping based on page row counts and offsets
-          auto const total_pages_in_this_column = page_row_counts[col_idx].size();
-
-          auto const page_indices = make_page_indices_async(
-            page_row_counts[col_idx], page_row_offsets[col_idx], total_rows, streams[col_idx]);
-
-          // Device vector to hold page indices with at least one required row
-          rmm::device_uvector<size_type> select_page_indices(total_rows, streams[col_idx], mr);
-
-          // Copy page indices with at least one required row
-          auto const filtered_pages_end_iter = thrust::copy_if(
-            rmm::exec_policy_nosync(streams[col_idx]),
-            page_indices.begin(),
-            page_indices.end(),
-            thrust::counting_iterator<size_type>(0),
-            select_page_indices.begin(),
-            is_row_required_fn{row_mask.nullable(), row_mask.null_mask(), row_mask.data<bool>()});
-
-          // Remove duplicate page indices across (presorted) rows
-          auto const filtered_uniq_page_end_iter =
-            thrust::unique(rmm::exec_policy_nosync(streams[col_idx]),
-                           select_page_indices.begin(),
-                           filtered_pages_end_iter);
-
-          // Number of final filtered pages for this column
-          size_t const num_surviving_pages_this_column =
-            thrust::distance(select_page_indices.begin(), filtered_uniq_page_end_iter);
-
-          total_surviving_pages.fetch_add(num_surviving_pages_this_column);
-
-          // Copy the filtered page indices for this column to host
-          auto host_select_page_indices = cudf::detail::make_host_vector(
-            cudf::device_span<cudf::size_type const>{select_page_indices.data(),
-                                                     num_surviving_pages_this_column},
-            streams[col_idx]);
-
-          // Vector to data page mask the this column
-          auto valid_pages = std::vector<bool>(total_pages_in_this_column, false);
-          std::for_each(host_select_page_indices.begin(),
-                        host_select_page_indices.end(),
-                        [&](auto const page_idx) { valid_pages[page_idx] = true; });
-
-          return valid_pages;
-        }));
+      // Construct a row indices mapping based on page row counts and offsets
+      auto const total_pages_in_this_column = page_row_counts[col_idx].size();
+
+      auto const page_indices = make_page_indices_async(
+        page_row_counts[col_idx], page_row_offsets[col_idx], total_rows, stream);
+
+      // Device vector to hold page indices with at least one required row
+      rmm::device_uvector<size_type> select_page_indices(total_rows, stream, mr);
```

Comment:
Mostly whitespace changes here. Click on gear icon above and check "Hide Whitespaces". The only diff is that the use of thread pool here has been removed. It was causing a race conditions/illegal memory access for large number of columns (or threads) and rows possibly due to memory overflow when creating so many `page_indices` and/or `select_page_indices`

Reply:
do you think we can keep this single-threader long term?